src/test/java/com/synopsys/integration/detect/ApplicationUpdaterTest.java

```diff
@@ -1,44 +1,150 @@
-        Request request = Mockito.mock(Request.class);
-        IntHttpClient intHttpClient = Mockito.mock(IntHttpClient.class);
-        Response response = Mockito.mock(Response.class);
-        Mockito.when(response.getStatusCode()).thenReturn(200);
-        Mockito.when(response.getHeaderValue(ApplicationUpdater.DOWNLOAD_VERSION_HEADER)).thenReturn("8.10.0");
-        Mockito.when(intHttpClient.execute(request)).thenReturn(response);
-        
-        DetectInfo detectInfo = Mockito.mock(DetectInfo.class);
-        Mockito.when(detectInfo.getDetectVersion()).thenReturn("8.9.0");
-        ApplicationUpdaterUtility applicationUpdaterUtility = Mockito.mock(ApplicationUpdaterUtility.class);
-        ApplicationUpdater applicationUpdater = new ApplicationUpdater(applicationUpdaterUtility, args);
-        Mockito.when(applicationUpdaterUtility.getIntHttpClient(true, ProxyInfo.NO_PROXY_INFO)).thenReturn(intHttpClient);
-        
-        boolean selfUpdated = applicationUpdater.selfUpdate();
-        Assertions.assertTrue(selfUpdated);
+    
+    private final String[] failureArgs = new String[] {
+            "-jar",
+            "/fake/path/to/synopsys-detect-n.n.n.jar",
+            "--blackduck.trust.cert=true", 
+            "--blackduck.api.token=dummyToken",
+            "--detect.tools=DETECTOR"};
+    
+    private final String[] successArgsWithValidProxy = new String[] {
+            "-jar",
+            "/fake/path/to/synopsys-detect-n.n.n.jar",
+            "--blackduck.url=".concat(fakeUrl),
+            "--blackduck.trust.cert=true", 
+            "--blackduck.api.token=dummyToken",
+            "--blackduck.proxy.host=".concat(fakeUrl),
+            "--blackduck.proxy.port=8080",
+            "--detect.tools=DETECTOR"};
+    
+    private final String[] successArgsWithInvalidProxy = new String[] {
+            "-jar",
+            "/fake/path/to/synopsys-detect-n.n.n.jar",
+            "--blackduck.url=".concat(fakeUrl),
+            "--blackduck.trust.cert=true", 
+            "--blackduck.api.token=dummyToken",
+            "--blackduck.proxy.host=".concat(fakeUrl),
+            "--detect.tools=DETECTOR"};
+    
+    @Test
+    public void testCanSelfUpdate() {
+        Assertions.assertTrue(new ApplicationUpdater(utility, successArgs).canSelfUpdate());
+    }
+    
+    @Test
+    public void testCanSelfUpdateWithoutBdUrl() {
+        Assertions.assertFalse(new ApplicationUpdater(utility, failureArgs).canSelfUpdate());
+    }
+    
+    @Test
+    public void testCanSelfUpdateWithSystemEnvBdUrl() {
+        ApplicationUpdaterUtility mockedUtility = Mockito.mock(ApplicationUpdaterUtility.class);
+        Mockito.when(mockedUtility.getSysEnvProperty(ApplicationUpdater.SYS_ENV_PROP_BLACKDUCK_URL)).thenReturn(fakeUrl);
+        Assertions.assertTrue(new ApplicationUpdater(mockedUtility, failureArgs).canSelfUpdate());
+    }
+    
+    @Test
+    public void testCanSelfUpdateWithSystemEnvDetectSrc() {
```

Comment:
Test to check for use of script related System environment properties: Detect Source.